File: api/src/common/entity/season.ts
import { Column, CreateDateColumn, Entity, ManyToOne, OneToMany, PrimaryGeneratedColumn } from 'typeorm';

import { Episode } from './episode';
import { Show } from './show';

@Entity()
export class Season {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ unique: true, nullable: true })
  providerId: string;

  @Column()
  number: number;

  @Column({ nullable: true })
  premiered?: Date;

  @Column({ nullable: true })
  ended?: Date;

  @Column({ nullable: true })
  network?: string;

  @Column({ nullable: true })
  episodeOrder?: number;

  @ManyToOne(() => Show, show => show.seasons)
  show?: Show;

  @OneToMany(() => Episode, episode => episode.season, {
    cascade: true,
  })
  episodes?: Episode[];

  @CreateDateColumn()
  createdAt?: Date;
}

File: api/models/db/season.go
package modelsdb

import (
	"time"
)

type Season struct {
	Model
	ProviderID   *string    `json:"providerId"`
	Number       uint8      `json:"number" gorm:"uniqueIndex:idx_season_uni;not null"`
	Premiered    *time.Time `json:"premiered"`
	Ended        *time.Time `json:"ended"`
	Network      *string    `json:"network"`
	EpisodeOrder *uint16    `json:"episodeOrder"`
	ShowID       uint       `json:"showId" gorm:"uniqueIndex:idx_season_uni"`
	Show         Show       `json:"-"`
	Episodes     []Episode  `json:"episodes"`
}
